### DISCUSSION TITLE
Critical Rendering Path: Inlining JS / CSS

### DISCUSSION THREAD

**Discussion by LukasBombach** [vercel/next.js#10838] [Help]
Hey there!

According to [best practices of performance on the web](https://www.smashingmagazine.com/2020/01/front-end-performance-checklist-2020-pdf-pages/) you will need some `inline JavaScript` and / or `inline CSS` at the top of your page to achieve a [critical rendering path](https://developer.mozilla.org/en-US/docs/Web/Performance/Critical_rendering_path).

I was looking for mechanisms or best practices to inline JS / CSS (at the top of your page) in `next.js` to conform that pattern, but I could not find any.

So I am not sure if there has already been done some work on this or if there are paradigms or common practices for the critical rendering path, esp. inline (critical) JS / CSS so I'd like to start this discussion here.

**Comment by Sangdt** [comment]
idk about inline js but for css you can check this out [this issue](https://github.com/zeit/next-plugins/issues/238), it might not work for the latest nextjs version.

**Comment by LukasBombach** [comment]
Hey everyone,

so we have tried to tackle this issue ourselves and this is what we've come up with:

https://github.com/stroeer/next-critical
